objectherkenning_openbare_ruimte/data_sampling/decos_sampling.py

```diff
@@ -0,0 +1,306 @@
+        decos_gdf = decos_helper.load_and_combine_decos(decos_folder)
+        self.frame_gdf = DecosSampling.add_metadata_and_permits(
+            DecosSampling.create_frame_gdf(input_folder),
+            metadata_folder,
+            decos_gdf,
+            decos_buffer,
+        )
+
+    @staticmethod
+    def load_metadata_gdf(path: str) -> gpd.GeoDataFrame:
+        """
+        Load a CSV metadata file as GeoDataFrame. Assumes geometry information is
+        stored in columns named gps_lon and gps_lat. CRS will be converted to Rijksdriehoek.
+
+        Parameters
+        ----------
+        path: str
+            Path to metadata SCV file
+
+        Returns
+        -------
+        GeoDataFrame containing the metadata
+        """
+        meta_df = pd.read_csv(path)
+        return gpd.GeoDataFrame(
+            meta_df,
+            geometry=gpd.points_from_xy(
+                x=meta_df.gps_lon,
+                y=meta_df.gps_lat,
+                crs=LAT_LON_CRS,
+            ),
+        ).to_crs(RD_CRS)
+
+    @staticmethod
+    def add_metadata_and_permits(
+        frame_gdf: gpd.GeoDataFrame,
+        metadata_folder: str,
+        decos_gdf: gpd.GeoDataFrame,
+        decos_buffer: float,
+    ) -> gpd.GeoDataFrame:
+        """
+        Add metadata and permit information for each image frame in frame_gdf.
+
+        The frame_gdf is expected to contain one line for each frame, and a column
+        "frame_src" indicating the video the frame originated from. This will be
+        used as key to locate the right metadata file.
+
+        The metadata file is expected to contain geometry (gps_lat, gps_lon)
+        and timestamps for each frame.
+
+        The permit data will be queried based on the time stamp and location.
+
+        Parameters
+        ----------
+        frame_gdf: GeoDataFrame
+            GDF with one row for each image frame.
+        metadata_folder: str
+            Location of metadata CSV files corresponding to the image frames.
+        decos_gdf: gpd.GeoDataFrame
+            Dataframe with permit data.
+        decos_buffer: float
+            Area around the permit location that will be considered as matching.
+
+        Returns
+        -------
+        Copy of the input frame_gdf with added metadata.
+        """
+        frame_gdf = frame_gdf.copy()
+
+        for src_name in frame_gdf["frame_src"].unique():
+            # src_name is the name of the source video file
+            print(f"Adding metadata for {src_name}")
+            idx = np.where(frame_gdf["frame_src"] == src_name)[0]
+
+            # get the identifying part of the file name and construct the metadata file path
+            meta_key_split = src_name.split(sep="-", maxsplit=2)
+            meta_file = f"{meta_key_split[0]}-{meta_key_split[2]}.csv"
+            meta_path = os.path.join(metadata_folder, meta_file)
+
+            # TODO do we want to use gps_time instead?
```

Comment:
Is this TODO solved?

Reply:
I changed it into a note for future reference.